docs: self-managed-CRDs (#660)
elastisys/compliantkubernetes: docs/operator-manual/user-manged-crds.md

# Self-managed cluster resources (Preview)

Adding certain cluster-wide resources, such as CustomResourceDefinitions, are cluster-wide settings that are prohibited by Compliant Kubernetes [for security purposes](../user-guide/demarcation.md).
However, many popular applications, in particular ones with Operators, require cluster-wide resources to be installed.
This preview feature enables application developers to self-manage a **predefined** set of such cluster-wide resources.
This trade-off means that application developers get the ability to install and manage such popular applications, but without compromising the security posture of the cluster.
By enabling this feature, a predefined set of cluster-wide resources is allowed to be installed.
In addition to CustomResourceDefinitions, other cluster-wide resources, such as the required ClusterRoles and ServiceAccounts are installed.
The list of currently supported applications for self-management is:

- SealedSecrets
- MongoDB

## Limitations

> Compliantkubernetes does not allow application developers to install ClusterRoles and ClusterRoleBindings.

Reviewer: ```suggestion
Compliant Kubernetes does not allow application developers to install ClusterRoles and ClusterRoleBindings.
```

For pre-approved operators that require ClusterRoles and ClusterRoleBindings.
They are installed during enabling of self-managed-CRDs feature.

> The installation of self-managed cluster resources is limited to a pre-defined Namespace.

Reviewer: Is this the same as the Namespaces of the application developer? If yes, I'd write:

```suggestion
The installation of self-managed cluster resources is limited to the Namespace for Application Developers.
```

Author: Not really, for example there exists a rolebinding that only exists in the sealedsecrets namespace. So sealedsecrets is required to be installed in the sealedsecrets namespace to work (predefined). I could install the rolebinding in a root namespace so it can be propagated via hnc to their other namespaces.

> To install into another Namespace, please contact your platform administrator.

Reviewer: Everything under `operator-manual` is meant to be read by platform administrators. Hence, we should remove this sentence.

## Enable self-managed cluster-wide resources

By default, this feature is disabled.
To enable it, add the following snippet to the environments `wc-config.yaml` file

```yaml
opa:
  # Enable feature
  restrictUserCRDs:
    enabled: true
    enforcement: deny
  # Enable Sealed Secrets
  sealedSecrets:
    enabled: true
  # Enable MongoDB
  mongodb:
    enabled: true
  # Add extra allowed CRDs
  userCRDs:
    allowedCRDs: []

> ```

Reviewer: I love how short and sweat the activation of this feature is. Nice! We seem to abstract a lot of complexity here.
